flink-core/src/main/java/org/apache/flink/api/common/typeutils/base/LocalDateComparator.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.common.typeutils.base;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.common.typeutils.TypeComparator;
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.DataOutputView;
+import org.apache.flink.core.memory.MemorySegment;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.time.LocalDate;
+
+/**
+ * This class can not extend {@link BasicTypeComparator}, because LocalDate is a
+ * Comparable of ChronoLocalDate instead of Comparable of LocalDate.
+ */
+@Internal
+public final class LocalDateComparator extends TypeComparator<LocalDate> implements Serializable {
+
+	private transient LocalDate reference;
+
+	protected final boolean ascendingComparison;
+
+	// For use by getComparators
+	@SuppressWarnings("rawtypes")
+	private final LocalDateComparator[] comparators = new LocalDateComparator[] {this};
+
+	public LocalDateComparator(boolean ascending) {
+		this.ascendingComparison = ascending;
+	}
+
+	@Override
+	public int hash(LocalDate value) {
+		return value.hashCode();
+	}
+
+	@Override
+	public void setReference(LocalDate toCompare) {
+		this.reference = toCompare;
+	}
+
+	@Override
+	public boolean equalToReference(LocalDate candidate) {
+		return candidate.equals(reference);
```

Comment:
use Objects.equals to avid NPE?

Reply:
`candidate` of `equalToReference` should not null.
Just like other comparator do.